spec/requests/sidebar_sections_controller_spec.rb

```diff
@@ -0,0 +1,163 @@
+# frozen_string_literal: true
+
+RSpec.describe SidebarSectionsController do
+  let(:admin) { Fabricate(:admin) }
+  fab!(:user) { Fabricate(:user) }
+
+  describe "#create" do
+    it "is not available for anonymous" do
+      post "/sidebar_sections.json",
+           params: {
+             title: "custom section",
+             links: {
+               "0" => {
+                 name: "categories",
+                 value: "/categories",
+               },
+               "1" => {
+                 name: "tags",
+                 value: "/tags",
+               },
+             },
+           }
+
+      expect(response.status).to eq(403)
+    end
+
+    it "creates custom section for user" do
+      sign_in(user)
+      post "/sidebar_sections.json",
+           params: {
+             title: "custom section",
+             links: {
+               "0" => {
+                 name: "categories",
+                 value: "/categories",
+               },
+               "1" => {
+                 name: "tags",
+                 value: "/tags",
+               },
+             },
+           }
+
+      expect(response.status).to eq(200)
+
+      expect(SidebarSection.count).to eq(1)
+      sidebar_section = SidebarSection.last
+
+      expect(sidebar_section.title).to eq("custom section")
+      expect(sidebar_section.user).to eq(user)
+      expect(sidebar_section.sidebar_urls.count).to eq(2)
+      expect(sidebar_section.sidebar_urls.first.name).to eq("categories")
+      expect(sidebar_section.sidebar_urls.first.value).to eq("/categories")
```

Comment:
For completeness sake, I believe we need to verify all the `sidebar_urls` records instead of just the first one.